bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteFlag.java

```diff
@@ -0,0 +1,72 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.api;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.EnumSet;
+import lombok.Getter;
+
+/**
+ * Flags to specify the behaviour of writes
+ */
+@Getter
+public enum WriteFlag {
+
+    /**
+     * Writes will be ackknowledged by the server even if the entry
+     * has not been persisted durably.
+     */
+    DEFERRED_FORCE(0x1 << 0);
```

Comment:
I think DEFERRED_SYNC may make more sense. I really don't know what FORCE means.
Generally in Linux O_SYNC flag is what signifies persistence on return, here it is deferred.

Reply:
I saw so many comments on this topic.
I like the word "sync" it is more "filesystem stuff" foe me.
The new API will be "force()" as in Java NIO API (FileChannel.force).

Honestly I would prefer something like "EXPLICIT_FORCE", because with this flag you have to call explicitly "WriteHandle#force"

I know this is very important because we cannot change this name in the future, but I would like to move forward

@jvrao @ivankelly @sijie

Reply:
I know `force` is used in java NIO api. but personally I am +1 on `DEFERRED_SYNC` (coming from a filesystem background). I am in-favor of linux filesystem flag names.

Reply:
renamed to DEFERRED_SYNC as decided on slack